src/api.rs

```diff
@@ -1438,3 +1507,57 @@ impl Handle {
         Ok(())
     }
 }
+
+// Struct wrapping an [IAudioEffectsManager](https://learn.microsoft.com/en-us/windows/win32/api/audioclient/nn-audioclient-iaudioeffectsmanager).
+pub struct AudioEffectsManager {
+    manager: IAudioEffectsManager,
+}
+
+impl AudioEffectsManager {
+    /// Gets the current list of audio effects for the associated audio stream.
+    pub fn get_audio_effects(&self) -> WasapiRes<Option<Vec<AUDIO_EFFECT>>> {
+        let mut audio_effects: *mut AUDIO_EFFECT = std::ptr::null_mut();
+        let mut num_effects: u32 = 0;
+
+        unsafe {
+            self.manager
+                .GetAudioEffects(&mut audio_effects, &mut num_effects)?;
+        }
+
+        if num_effects > 0 {
+            let effects_slice =
+                unsafe { slice::from_raw_parts(audio_effects, num_effects as usize) };
+            let effects_vec = effects_slice.to_vec();
+            // Free the memory allocated for the audio effects.
+            unsafe { CoTaskMemFree(Some(audio_effects as *mut _)) };
+            Ok(Some(effects_vec))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+/// Struct wrapping an [AcousticEchoCancellationControl](https://learn.microsoft.com/en-us/windows/win32/api/audioclient/nn-audioclient-iacousticechocancellationcontrol).
+pub struct AcousticEchoCancellationControl {
+    control: IAcousticEchoCancellationControl,
+}
+
+impl AcousticEchoCancellationControl {
+    /// Sets the audio render endpoint that should be used as the reference stream for acoustic echo cancellation (AEC).
+    /// Setting the value to None will result in Windows using its own algorithm to pick the loopback reference device.
```

Comment:
This text would be more helpful if it mentioned where to get the id from. Could you add a link to the method to call?

Reply:
Done.